Feature/getting started improvements

hartzell opened the pull request:
I found the quick start frustrating because I had no idea how to create an image.

I eventually found a simple recipe in the repo's README.md.

This PR duplicates it in the quick start section.  It's CentOS specific but it gives pointers to additional info and discussion.

Also, minor edits so that all of the examples showed similar shell prompts.

Commit:
Quick hints to build a CentOS 7 image.

It was frustrating to try to start quickly via the Quick Start and not
have any idea where to find an image.  There are hints in the README in
the repo, but it's convenient to have them here.

I pulled the content from the README.md and referenced it and the
tutorials section.

"Someone" should probably work up a posse of examples for the Linux
Flavors du jour.

Commit:
Add shell prompts to the examples

Make the examples consistent-ish with the examples in the README.md by
adding the traditional '$' shell prompt where appropriate.

Commit:
Fix broken link to README.md

Commit:
Add prompts to the Installation quick start

As in the earlier commit, add shell prompts to the lines in the example,
making them consistent with other examples here and in the README.md.

Commit:
Fix image name (quickstart differs from README.md)

Commit:
Wordsmithing

vsoch commented on pages/docs/overview/start.md:
```diff
@@ -9,19 +9,42 @@ toc: false
+
+It's easiest to build an image on a "compatible" host.  Here's a quick
+recipe to build a CentOS 7 image on a CentOS host.  See
+the [bootstrapping section][readme-bootstrapping] of the README for
```
you'll have to put the full URL to the README here, this will likely lead to a 404.

vsoch commented on pages/docs/overview/start.md:
```diff
@@ -9,19 +9,42 @@ toc: false
+recipe to build a CentOS 7 image on a CentOS host.  See
+the [bootstrapping section][readme-bootstrapping] of the README for
+the full story and the [tutorials] for deeper advice.
+
```
ditto here I think

vsoch commented on pages/docs/overview/start.md:
````diff
@@ -9,19 +9,42 @@ toc: false
+   MirrorURL: http://mirror.centos.org/centos-%{OSVERSION}/%{OSVERSION}/os/$basearch/
+   Include: yum
+   ```
+
````
I really like this change to the Quickstart, but instead of doing Centos 7, could we do a technology that more users are familiar with, Docker? So provide an example of `Bootstrap: docker` and then maybe point the user to the folder of example specs for others.

vsoch commented on pages/docs/overview/start.md:
```diff
@@ -32,35 +55,37 @@ Singularity.Centos7.img> exit
 
 {% include links.html %}
+
+[readme-bootstrapping]: https://github.com/singularityware/singularity/blob/master/README.md#bootstrapping-new-images
```
oh oups, disregard my previous comment... forgot about this :)

hartzell commented:
I built a CentOS image because the rest of the examples on that page refer to a CentOS image.

I think that you're suggesting above that instead of providing an example that uses `bootstrap` to build the CentOS image (which only works easily on a CentOS host) I might provide an example that builds a Quick Start CentOS image by importing `docker://centos`. 

I can see how that would be more generically useful.  I'll rework this tonight.

vsoch commented:
Sweet! Yes I love it. Showing after that the bootstrap spec would still be ok, because the user would see that there are many ways to skin a cat.

The common use of centOS is because there is this weird guy (cough cough @gmkurtzer) that like, made it or something? He's wandering the halls of documentation for Singularity and putting it everywhere... we should find him and give him some oatmeal or something. :p

hartzell commented:
I'll see what I can whip up over the weekend.

I'm all about CentOS at work (and FreeBSD when I let my hair down) but I do like to keep an open mind....

vsoch commented:
I hope your mind is open... my go to is Ubuntu! I'm pretty sure it's just because it's purple. :purple_heart:

Commit:
Rework example to pull image from Docker Hub

Pulling an existing image from Docker Hub will make it easier for a
wider range of Linux users to get going quickly.

Rework the demo to import a CentOS 7 image.  Give a few other popular
Linuxes a shout out and refer to the Singularity and Docker page.

hartzell commented:
[Beastie's usually red](http://www.za.freebsd.org/art.html), but it's open minded too... 😈 .

This update calls out some of the other Linuxes too.

vsoch commented:
hey @hartzell do you still want to make these changes? If you want to rebase with master (or PR to docs/2.4 is relevant) we can revisit.

hartzell commented:
Hi @vsoch -- Yes, I'll pick this back up.  I'm working on this sporadically for now.  I appreciate your patience.

vsoch commented:
Of course! There really isn't a rush, I just wanted to touch bases. Chat when we do! :)
